qiskit/circuit/library/n_local/n_local.py

```diff
@@ -134,6 +135,9 @@ def __init__(
         self._data = None
         self._bounds = None
 
+        if not isinstance(reps, int):
```

Comment:
This is quite a strict check since it excludes other common integer types, such as `np.int`s. I think we can be quite flexible here, so how about replacing this by something like
```
if int(reps) != reps:  
```
which works for numbers like `3, np.int32(3), np.int64(3), 3.0`. 

If you think we should be more strict, we could also just add other integer types in the typecheck.

Reply:
Alright I am looking into it, thanks.